README.rst

```diff
@@ -454,16 +454,184 @@ template ``context_processors`` in ``settings.py`` as shown below.
                 ],
                 'context_processors': [
                     # ... other context processors ...
-                    'openwisp_utils.admin_theme.context_processor.menu_items'    # <----- add this
+                    'openwisp_utils.admin_theme.context_processor.menu_groups'    # <----- add this
                 ],
             },
         },
     ]
 
-If you need to define custom menu items, see:
-`OPENWISP_ADMIN_MENU_ITEMS <#openwisp_admin_menu_items>`_.
+``register_menu_group``
+^^^^^^^^^^^^^^^^^^^^^^^^
 
-Users will only be able to see menu items for objects they have permission to either view or edit.
+It allow us to register different menu item or group at different position in the ``Main Navigation Menu``.
+
+**Syntax:**
+
+.. code-block:: python
+
+    register_menu_group(position, config)
+
++--------------------+-------------------------------------------------------------+
+| **Parameter**      | **Description**                                             |
++--------------------+-------------------------------------------------------------+
+| ``position``       | (``int``) Position of the group or item.                    |
++--------------------+-------------------------------------------------------------+
+| ``config``         | (``dict``) Configuration of goup or item.                   |
++--------------------+-------------------------------------------------------------+
+
+Code example:
+
+.. code-block:: python
+
+    from openwisp_utils.admin_theme.menu import register_menu_group
+
+    my_group_items = {
+        1: {
+            'label': 'Add user',
+            'model': 'auth.User',
+            'name': 'changelist',
+            'icon': 'add-icon',
+        },
+        2: {
+            'label': 'Add user',
+            'model': 'auth.User',
+            'name': 'add',
+            'icon': 'add-icon',
+        },
+    }
+    register_menu_group(position=1, config={'label': 'My Group', 'items': my_group_items})
+    register_menu_group(
+        position=2,
+        config={
+            'model': 'test_project.Shelf',
+            'name': 'changelist',
+            'label': 'View Shelf',
+            'icon': 'shelf-icon',
+        },
+    )
+    register_menu_group(
+        position=3, config={'label': 'Link Name', 'url': 'https://link.com'}
+    )
+
+It is recommended to use ``register_menu_group`` in the ``ready`` method of the AppConfig.
+
+Note: ``register_menu_items`` is deprecated by ``register_menu_group`` and will be removed in the
+future versions. ``register_menu_items`` will be shown at the top of navigation menu and above 
+any register_menu_group items.
+
+``Create a general link``
```

Comment:
This is not code, therefore it should not be formatted with the double slant quotes around.